src/app/operations/planning/planning.component.ts

```diff
@@ -0,0 +1,93 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { OperationDbService } from '../operations-db.service';
+import { AreaDbService } from '../../area/area-db.service';
+import { Operation} from '../../model/operation';
+import { Area } from '../../model/area';
+import * as moment from 'moment';
+import { VirtualScrollerComponent } from 'ngx-virtual-scroller';
+
+@Component({
+  selector: 'app-planning',
+  templateUrl: './planning.component.html',
+  styleUrls: ['./planning.component.scss']
+})
+export class PlanningComponent implements OnInit {
+
+  @ViewChild(VirtualScrollerComponent)
+  private virtualScroller: VirtualScrollerComponent;
+
+  operation: Operation;
+  operationsStored: Operation[];
+  parentAreas: String[][];
+  currentId = null;
+  tempDate = '3000-12-12';
+
+  constructor(private operationDbService: OperationDbService, private areaDbService: AreaDbService) {
+      this.operation = new Operation();
+      this.operation.area = new Area();
+      this.parentAreas = [];
+   }
+
+  ngOnInit() {
+    this.operationDbService.getOperations().then((operations) => {
+      this.operationsStored = operations;
+
+      this.sortByDates();
+
+      this.formatDatesAndGetParentAreas();
+
+      this.virtualScroller.scrollToIndex(this.currentId);
+    });
+  }
+
+  /**
+   * sorts all the operations by their date
+   * if a date already exists, the date is set to null so an operation isn't displayed twice
+   */
+  sortByDates() {
```

Comment:
You can set this and the other function as `private` to indicate that it's only used within the component